Login/loginRouter.js

```diff
@@ -0,0 +1,55 @@
+const express = require("express");
+const router = express.Router();
+const DB = require("./loginHelper.js");
+const jwt = require("jsonwebtoken");
+const bcrypt = require("bcrypt");
+const secret = require("../secret.js");
+
+// **** BELOW IMPLEMENTED IN USERS ROUTER ****
+// router.get("/", async (req, res) => {
+//   try {
+//     const users = await DB.getUsers();
+//     res.status(200).json(users);
+//   } catch (err) {
+//     res.status(400).json({ success: false, err });
+//   }
+// });
+
+router.post("/", (req, res) => {
+  let { username, password } = req.body;
+  DB.findUser(username).then(user => {
+    if (user && bcrypt.compareSync(password, user.password)) {
+      const token = generateToken(user);
+      console.log(token);
+      res.status(200).json({ message: `Welcome, ${user.username}!`, token });
+    } else {
+      res
+        .status(500)
```

Comment:
error 500 is a server error. You would want to put the server error inside of a .catch statement. 

Your else conditional would be handling the case where the user didn't provide all required fields. Which would be a 400 error.

Reply:
Your else conditional could also run if the user provided the incorrect username or password.

Reply:
Also, it does run if the user doesn't exist. Long story short, I'd like to see a bit more separation in your endpoint to clearly define each case. (user provides correct information, user provides incorrect information, user does not provide a password, and server error)